### PR TITLE
445 improve docs to better outline the tome creation process

### LINKED ISSUE #445: Improve docs to better outline the tome creation process
**Describe the bug**
One of the main ways people will get involved with developing on Realm is through tomes.
Currently new developers are having a hard time with the tome development workflow.

Add a page to the docs that describes how to create a new tome.

**To Reproduce**

**Expected behavior**

**Screenshots**

**Desktop (please complete the following information):**

**Smartphone (please complete the following information):**

**Additional context**

### PR DESCRIPTION
**Pull request by hulto**
#### What type of PR is this?
/kind documentation

#### What this PR does / why we need it:
Updates docs around tome creation.

#### Which issue(s) this PR fixes:
Fixes #445

### CODE REVIEW

File: docs/_docs/user-guide/golem.md [hunk 1]
```diff
@@ -24,6 +24,126 @@ cargo build --release && \
...
+print("\n")
+print("\n")
+```
+
+### Design ideology
+
+Tomes are designed to accomplish specific workflows.
+Some small tomes have been created to accomplish basic sysadmin tasks like reading, writing files, checking processes etc.
+Ideally though if you have a specific workflow you'll define it as a tome.
+We want to avoid queueing multiple questst to accomplish a workflow.
```
Comment (Cictrone) [inline docs/_docs/user-guide/golem.md:126]: nit: quests

File: docs/_docs/user-guide/golem.md [hunk 2]
```diff
@@ -24,6 +24,126 @@ cargo build --release && \
...
+
+### Testing your tome
+
+Now that your tome is created lets test it locally.
```
Comment (Cictrone) [inline docs/_docs/user-guide/golem.md:130]: nit: let's

**Review by Cictrone — APPROVED** [review]
just some nits

File: docs/_docs/user-guide/eldritch.md [hunk 3]
```diff
@@ -14,6 +14,8 @@ By embedding the interpreter into the agent conditional logic can be quickly eva
 
 Eldritch is currently under active development to help delineate methods in development the description contains the phrase `X method will`.
 
+**Trying to create a new tome check out the guide in [Golem](/user-guide/golem)**
```
Comment (KCarretto) [inline docs/_docs/user-guide/eldritch.md:17]: nit: Trying to create a tome? Check out the guide in ...

File: docs/_docs/user-guide/golem.md [hunk 4]
```diff
@@ -24,6 +24,126 @@ cargo build --release && \
     ../target/debug/golem ../../tests/golem_cli_test/tomes/hello_world.tome
 ```
 
+## Creating and testing tomes
+
+Golem is a great way to create and test tomes without deploying the entire Realm project.
```
Comment (KCarretto) [inline docs/_docs/user-guide/golem.md:29]: Golem is a great way to quickly develop and iterate tomes without needing to deploy an agent and wait for callbacks. It enables a more interactive experience for Eldritch, which helps tome developers move quickly.

File: docs/_docs/user-guide/golem.md [hunk 5]
```diff
@@ -24,6 +24,126 @@ cargo build --release && \
...
+- On a specific tome `/path/to/tome/eldritch.main`
+
+In this guide we'll leverage the last one.
+
+### Tome structure
+
+Copy an existing tome and rename it to your desired name.
+
+```bash
+[~/realm]$ cd tavern/tomes/
```
Comment (KCarretto) [inline docs/_docs/user-guide/golem.md:43]: Maybe we should add a `golem init` command, to build the boiler plate for new tomes

File: docs/_docs/user-guide/golem.md [hunk 6]
```diff
@@ -24,6 +24,126 @@ cargo build --release && \
...
+### Tome Metadata
+
+Your tome Metadata should look like this:
+
+```yaml
+name: List files # The name of your tome.
+description: List the files and directories found at the path # A description to help users understand what your tome does.
+author: hulto # Your Github username.
+support_model: FIRST_PARTY # Is this a tome that ships with Realm or not?
+tactic: RECON # What stage of the MITRE ATT&CK chain does this tome fall into?
```
Comment (KCarretto) [inline docs/_docs/user-guide/golem.md:72]: Which MITRE ATT&AK tactic best describes this tome?

File: docs/_docs/user-guide/golem.md [hunk 7]
```diff
@@ -24,6 +24,126 @@ cargo build --release && \
...
+./new_tome/
+├── files
+│   └── test-file
+├── main.eldritch
+└── metadata.yml
+```
+
+_If you have no files associated with your tome you can ommit the `files` dir_
+
+- main.eldritch - Defines our actual eldritch functionality and what will be executed on the agent.
```
Comment (KCarretto) [inline docs/_docs/user-guide/golem.md:60]: nit: Eldritch

File: docs/_docs/user-guide/golem.md [hunk 8]
```diff
@@ -24,6 +24,126 @@ cargo build --release && \
...
+support_model: FIRST_PARTY # Is this a tome that ships with Realm or not?
+tactic: RECON # What stage of the MITRE ATT&CK chain does this tome fall into?
+paramdefs: # A list of inputs the tome requires.
+- name: path # The name of the input parameter `input_params['path']`.
+  type: string # The type of the input parameter.
+  label: File path # A Label that will be displayed in the UI to help users understand the purpose.
+  placeholder: "/etc/" # A placeholder to give users an idea of how their input should be formatted.
+```
+
+### Building your eldritch script
```
Comment (KCarretto) [inline docs/_docs/user-guide/golem.md:80]: nit: Writing Eldritch*

File: docs/_docs/user-guide/golem.md (hunk 1 above)
Comment (KCarretto) [inline docs/_docs/user-guide/golem.md]: nit: quests

**Review by KCarretto — CHANGES_REQUESTED** [review]

File: docs/_docs/user-guide/golem.md (hunk 5 above)
Comment (hulto) [reply docs/_docs/user-guide/golem.md:43]: I don't love the idea of adding non operational stuff to golem and imix.
Instead we could add a template tome that users can copy out.

**Review by KCarretto — APPROVED** [review]

**Pull request merged**